vertical-pod-autoscaler/pkg/utils/vpa/api.go

```diff
@@ -87,6 +118,40 @@ func UpdateVpaStatusIfNeeded(vpaClient vpa_api.VerticalPodAutoscalerInterface, v
 	return nil, nil
 }
 
+// annotationsAreStaleAndChanged checks if any of the specified annotations are stale
+// (older than freshnessUpdateIntervalSeconds) or have changed between the new and old annotations.
+func annotationsAreStaleAndChanged(
+	newAnnotations, oldAnnotations map[string]string,
+	freshnessUpdateIntervalSeconds int,
+) bool {
+	keys := []string{"cpu_last_updated", "rss_last_updated", "jvm_heap_last_updated"}
+	staleThreshold := time.Now().Add(-time.Duration(freshnessUpdateIntervalSeconds) * time.Second)
+
+	for _, key := range keys {
+		newValue, newExists := newAnnotations[key]
+		oldValue, oldExists := oldAnnotations[key]
+
+		oldTime, oldErr := time.Parse(time.RFC3339, oldValue)
+		if (!oldExists || oldErr != nil || oldTime.Before(staleThreshold)) &&
```

Comment:
If there is `oldErr` we should log it here to signal that something unexpected is going on.